File: 백준/Bronze/2884. 알람 시계/알람 시계.py
hour, minute = map(int, input().split())

if minute < 45:
    if hour == 0:
        alarm_hour = 23
    else:
        alarm_hour = hour - 1
    alarm_minute = minute + 15
else:
    alarm_hour = hour
    alarm_minute = minute - 45

print(alarm_hour,alarm_minute)

File: 백준/Bronze/2884. 알람 시계/알람 시계.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int hour = sc.nextInt();
        int minute = sc.nextInt();
        sc.close();

        // 45분 일찍 알람 설정하기
        if (minute >= 45)
            minute = minute - 45;
        else {
            minute = 15 + minute;
            hour -= 1;
            if (hour < 0)
                hour = 23;
        }
        System.out.println(hour + " " + minute);

    }
}